django/db/backends/postgresql/schema.py

```diff
@@ -52,11 +64,10 @@ class DatabaseSchemaEditor(BaseDatabaseSchemaEditor):
     def quote_value(self, value):
         if isinstance(value, str):
             value = value.replace("%", "%%")
-        adapted = psycopg2.extensions.adapt(value)
-        if hasattr(adapted, "encoding"):
-            adapted.encoding = "utf8"
-        # getquoted() returns a quoted bytestring of the adapted value.
-        return adapted.getquoted().decode()
+        if self.connection.is_psycopg3:
+            return sql.quote(value, self.connection.connection)
+        else:
+            return sql.quote(value)
```

Comment:
It feels like `sql.quote` should be defined at module time take a required `connection` argument in `psycopg_any` so no branching needs to take place.